### PR TITLE
feat: add function to parse variable from string

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/systems/abstractsystem.jl [hunk 1]
```diff
@@ -3285,6 +3285,86 @@ function dump_unknowns(sys::AbstractSystem)
...
+Roughly supports the following CFG:
+
+```
+varname                  = "D(" varname ")" | arrvar | maybe_dummy_var
+arrvar                   = maybe_dummy_var "[idxs...]"
+idxs                     = int | int "," idxs
+maybe_dummy_var          = namespacedvar | namespacedvar "(t)" |
+                           namespacedvar "(t)" "ˍ" ts | namespacedvar "ˍ" ts |
+                           namespacedvar "ˍ" ts "(t)"
+ts                       = "t" | "t" ts
```
Comment (ChrisRackauckas) [inline src/systems/abstractsystem.jl:3301]: `nameof(independent_variable(sys))`?

File: src/systems/abstractsystem.jl [hunk 2]
```diff
@@ -3285,6 +3285,87 @@ function dump_unknowns(sys::AbstractSystem)
...
+
+Where `iv` is the independent variable, `int` is an integer and `ident` is an identifier.
+"""
+function parse_variable(sys::AbstractSystem, str::AbstractString)
+    iv = has_iv(sys) ? string(getname(get_iv(sys))) : nothing
+
+    # I'd write a regex to validate `str`, but https://xkcd.com/1171/
+    str = strip(str)
+    derivative_level = 0
+    while startswith(str, "D(") && endswith(str, ")")
```
Comment (ChrisRackauckas) [inline src/systems/abstractsystem.jl:3313]: We may need to validate that `D` is also the differential?

Comment (AayushSabharwal) [reply src/systems/abstractsystem.jl:3313]: `D` never shows up in expressions, it always prints as `Differential(t)(x(t))`. In the interest of concision, I assumed that `D` refers to the derivative with respect to the independent variable. I could also add a case that handles `Differential(t)(x(t))` and validates that the independent variable matches.

Comment (ChrisRackauckas) [reply src/systems/abstractsystem.jl:3313]: Okay

**Pull request merged**